village/src/App.js

```diff
@@ -11,14 +13,49 @@ class App extends Component {
       smurfs: [],
     };
   }
+
+  componentDidMount() {
+    axios
+      .get('http://localhost:3333/smurfs')
+      .then(response => {
+        this.setState({smurfs: response.data});
+      })
+      .catch(err => {
+        console.log(err);
+      })
+  }
+
+  addSmurf = (name, age, height) => {
+
+    const newSmurf = {
+      name: name,
+      age: age,
+      height: height
+    }
+
+    axios
+      .post('http://localhost:3333/smurfs', newSmurf)
+      .then(response => {
+        this.setState({ smurfs: response.data });
+      })
+      .catch(err => {
+        console.log(err);
+      })
+  }
   // add any needed code to ensure that the smurfs collection exists on state and it has data coming from the server
   // Notice what your map function is looping over and returning inside of Smurfs.
   // You'll need to make sure you have the right properties on state and pass them down to props.
   render() {
     return (
       <div className="App">
-        <SmurfForm />
-        <Smurfs smurfs={this.state.smurfs} />
+        <nav> 
+          <Navigation />
+        </nav>
+        <div className="container">
+          <Route exact path='/smurf-form' render = { props => <SmurfForm {...props} addSmurf={this.addSmurf} /> } />
+          <Route exact path='/' render= { props => <Smurfs {...props} smurfs={this.state.smurfs} /> } />
```

Comment:
For readability, I'd recommend breaking your routes down onto multiple lines like you did with other components. 

example:
```   
<input
       onChange={this.handleInputChange}
       placeholder="name"
       value={this.state.name}
       name="name"
/>
```